solutions/usaco-988.mdx

```diff
@@ -112,4 +115,65 @@ print(ans, file=open('photo.out', 'w'))
 ```
 
 </PySection>
+<JavaSection>
+	
+**Time Complexity:** $\mathcal{O}(N^2)$
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class Photoshoot {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("photo");
+
+		int N = io.nextInt();
+		int[] bessieList = new int[N];
+		int[] correctList = new int[N];
+		// Used to prevent repeated values in our permutation
+		Set<Integer> isItThere = new HashSet<>();
```

Comment:
boolean array? if i'm understanding correctly the values are in range [1, N], so no need for a hashset

Reply:
Wouldn't you need to iterate through every value less than k for the boolean array, to see if a given value has been true before? I used hashset because it's O(1) and has a quick and easy .contains call. But if there's a more efficient way to use a boolean array, I'd be happy to here it.

Reply:
when you use a number x (x is between 1 and n), set `itItThere[x] = true`. to check if x has been used before, check if `isItThere[x]` is true